File: python/udp_client.py
import sys
import time

from support_loader import espp

udp_client = espp.UdpSocket(espp.UdpSocket.Config(espp.Logger.Verbosity.debug))

# defined out here so that it's only initialized once, not on each callback
start = time.time()
def task_func() -> bool:
    global start
    global udp_client
    port = 5555
    ip = "127.0.0.1"
    send_config = espp.UdpSocket.SendConfig(
        ip, port
    )
    elapsed = time.time() - start
    print(f"[{elapsed:.3f}] Hello from task")
    udp_client.send("Hello world!\n", send_config)
    time.sleep(.5)
    return False # we don't want to stop the task

task = espp.Task(
    task_func,
    espp.Task.BaseConfig("test task")
)
task.start()

time.sleep(5)

sys.exit(0)

File: pc/tests/udp_client.cpp
#include "espp.hpp"

using namespace std::chrono_literals;

static auto start = std::chrono::high_resolution_clock::now();
static auto client = espp::UdpSocket({.log_level = espp::Logger::Verbosity::DEBUG});

bool task_func() {
  auto send_config = espp::UdpSocket::SendConfig{.ip_address = "127.0.01", .port = 5555};
  fmt::print("Sending message\n");
  client.send("Hello world\n", send_config);
  std::this_thread::sleep_for(500ms);
  return false; // don't want to stop the task
}

int main() {
  auto task = espp::Task(espp::Task::SimpleConfig{
      .callback = task_func, .task_config = espp::Task::BaseConfig{.name = "test_task"}});
  task.start();

  std::this_thread::sleep_for(5s);

  return 0;
}
